internal/middleware/memoryprotection/memory_protection.go

```diff
@@ -0,0 +1,91 @@
+package memoryprotection
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	middleware "github.com/grpc-ecosystem/go-grpc-middleware/v2"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	log "github.com/authzed/spicedb/internal/logging"
+)
+
+// RequestsProcessed tracks requests that were processed by this middleware.
+var RequestsProcessed = promauto.NewCounterVec(prometheus.CounterOpts{
+	Namespace: "spicedb",
+	Subsystem: "memory_middleware",
+	Name:      "requests_processed_total",
+	Help:      "Total requests processed by the memory protection middleware (flag --memory-protection-enabled)",
+}, []string{"endpoint", "accepted"})
```

Comment:
Does this give us anything that a request counter doesn't? Should we change this to be "requests rejected" or something?